sql/gen-sql-markdown.py

```diff
@@ -136,6 +137,27 @@ def _make_pretty_note(note):
         return "**Note:**\n%s\n" % note
 
 
+def _make_pretty_deprecated(deprecated):
+    """
+    Makes the deprecated description pretty and returns a formatted string if `deprecated`
+    is not an empty string. Otherwise, returns None.
+
+    Expected input:
+
+        ...
+
+    Expected output:
+    **Deprecated:**
+
+    ...
+
+    """
+
+    if deprecated != "":
+        deprecated = "\n".join(map(lambda n: n[4:], deprecated.split("\n")))
```

Comment:
@cloud-fan, for instance this logic matters.

If we put some strings in ..

```scala
@ExpressionDescription(
  ...
  deprecated = """
    blah blah blah
    I am super long line
    
    1. one
    2. I am markdwn

    - bullet marks
    - aa

    You know what? actually you can write a markdown table as well.

    | Tables        | Are           | Cool  |
    | ------------- |:-------------:| -----:|
    | col 3 is      | right-aligned | $1600 |
    | col 2 is      | centered      |   $12 |
  """)
```

`gen-sql-markdown.py` here split the doc into each line and then get rid of the first 4 characters for each. Last newline makes a proper indentation between each field.

Reply:
will people use multi-line string literal in Scala? e.g.
```
deprecated = """
  |xxx
  |xxx
""".stripMargin
```

Reply:
That cannot be used here due to compilation error. It seems annotation parameters only allow constants. There was a discussion about it before when I first add `examples`.

Reply:
what if the expression is inside an object? e.g.
```
object A {
  @ExpressionDescription(
    note = """
      blabla
    """ )
  class B
}
```
Then the string has 6 spaces at the beginning.

It looks to me that we can relax the format a little bit: just trim all the spaces at the beginning. Then we can even use a single line string.

Reply:
Yes, in that case it might be a problem although there look no case so far after this fix (I skimmed the built doc).

The thing we should take into account is we shouldn't trim all the white spaces all because spacing matters in md rendering. For instance, if we have a doc like ..

```
object A {
  @ExpressionDescription(
    note = """
      blabla
      - first
          blabla
      -second
          blabla
    """ )
  class B
}
```

We should only trim leading 6 spaces for each line. So, it will bring a bit of complication. It's doable though.

If you don't mind, I will handle it later. Was thinking it better have a single format for now. We should encourage to have more verbose and detailed doc anyway (rather than single line doc) in general.

Reply:
Yes, in that case it might be a problem although there look no case so far after this fix (I skimmed the built doc).

The thing we should take into account is we shouldn't trim all the white spaces all because spacing matters in md rendering. For instance, if we have a doc like ..

```
object A {
  @ExpressionDescription(
    note = """
      blabla
      - first
          blabla
      -second
          blabla
    """ )
  class B
}
```

We should only trim leading 6 spaces for each line. So, it will bring a bit of complication. It's doable though.

If you don't mind, I will handle it later. Was thinking it better have a single format for now. We should encourage to have more verbose and detailed doc anyway (rather than single line doc) in general.

Reply:
ah that makes sense, it's more tricky than I thought...